### PR TITLE
Gitignore ignoring rust+clion+all

### PR DESCRIPTION
**Pull request by bonomat**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,157 @@
...
+
+# Created by https://www.toptal.com/developers/gitignore/api/rust,clion+all,emacs
+# Edit at https://www.toptal.com/developers/gitignore?templates=rust,clion+all,emacs
+
+### CLion+all ###
+# Covers JetBrains IDEs: IntelliJ, RubyMine, PhpStorm, AppCode, PyCharm, CLion, Android Studio, WebStorm and Rider
+# Reference: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
+
+# User-specific stuff
+.idea/**/workspace.xml
```
Comment (D4nte) [inline .gitignore:10]: Well, actually, I would encourage you to add that to your own gitignore! https://stackoverflow.com/questions/5724455/can-i-make-a-user-specific-gitignore-file

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,157 @@
...
+# .idea/compiler.xml
+# .idea/jarRepositories.xml
+# .idea/modules.xml
+# .idea/*.iml
+# .idea/modules
+# *.iml
+# *.ipr
+
+# CMake
+cmake-build-*/
```
Comment (D4nte) [inline .gitignore:46]: Not really needed.

File: .gitignore [hunk 3]
```diff
@@ -1,2 +1,157 @@
...
+.idea/httpRequests
+
+# Android studio 3.1+ serialized cache file
+.idea/caches/build_file_checksums.ser
+
+### CLion+all Patch ###
+# Ignores the whole .idea folder and all .iml files
+# See https://github.com/joeblau/gitignore.io/issues/186 and https://github.com/joeblau/gitignore.io/issues/360
+
+.idea/
```
Comment (D4nte) [inline .gitignore:82]: So.. Delete the first 80 lines of this file?

**Comment by bonomat** [comment]
@D4nte : the whole file was auto generated using gitignore.io :)

**Pull request merged**